lib/ui.rb

```diff
@@ -90,6 +101,20 @@ def warn(s)
       def error(s)
         STDERR.puts s
       end
+
+      private
+
+      def show_progress
+        $stdout.tty? || ENV["SHOW_PROGRESS_OUTPUT"]
```

Comment:
Wouldn't it make sense to have a machinery specific prefix to the environment variable in order to avoid clashes with other programs: e.g. `MACHINERY_SHOW_PROGRESS_OUTPUT`.